server/src/test/java/org/elasticsearch/index/query/MultiMatchQueryBuilderTests.java

```diff
@@ -536,4 +564,18 @@ private static IndexMetaData newIndexMeta(String name, Settings oldIndexSettings
             .build();
         return IndexMetaData.builder(name).settings(build).build();
     }
+
+    private void assertQueryWithAllFieldsWildcard(Query query) {
+        assertEquals(DisjunctionMaxQuery.class, query.getClass());
+        DisjunctionMaxQuery disjunctionMaxQuery = (DisjunctionMaxQuery) query;
+        int noMatchNoDocsQueries = 0;
+        for (Query q : disjunctionMaxQuery.getDisjuncts()) {
+            if (q.getClass() == MatchNoDocsQuery.class) {
+                noMatchNoDocsQueries++;
+            }
+        }
+        assertEquals(11, noMatchNoDocsQueries);
```

Comment:
small suggestion: I think we have 11 clauses because that its the number of fields defined in AbstractBuilderTestCase#MAPPED_LEAF_FIELD_NAMES. To make the test more robust to future changes, maybe use that array size here?

Reply:
I thought about it but I think that's not correct, both `MAPPED_LEAF_FIELD_NAMES` and `MAPPED_FIELD_NAMES` contain less fields + some aliases and are not used for the registration of the fields. The registration takes place here: https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java#L388 and doesn't use either of those arrays.

Reply:
Bummer. And I cannot see another simple way to get to this value from the shard context eithter, other than using the same code that is used by the query builder, and that would defeat the purpose. Guess it needs to stay like that then.